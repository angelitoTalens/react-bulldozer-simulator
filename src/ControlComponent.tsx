import React from "react";
import "./styles/ControlComponent.scss"

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faRedoAlt, faUndoAlt, faPlayCircle } from "@fortawesome/free-solid-svg-icons";

import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Button from "react-bootstrap/Button";


type ControlComponentProps = {
    onClickRotateLeft: (event: React.MouseEvent<HTMLButtonElement>) => void,
    onClickRotateRight: (event: React.MouseEvent<HTMLButtonElement>) => void,
    onClickAdvance: (event: React.MouseEvent<HTMLButtonElement>) => void,
    onUserTerminate: (event: React.MouseEvent<HTMLButtonElement>) => void,
    onStepSettingChange: (event: React.FocusEvent<HTMLInputElement>) => void,
}

type ControlComponentState = {
    stepSetting: number;
}

class ControlComponent extends React.Component<ControlComponentProps, ControlComponentState> {

    state:ControlComponentState;

    constructor(props:ControlComponentProps) {
        super(props);
        this.state = {
            stepSetting: 1
        }
    }

    resetStepSetting() {
        let state = this.state;
        state.stepSetting = 1;
        this.setState(state);
    }

    handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
        let state = this.state;
        state.stepSetting = event.target.value ? Number(event.target.value) : NaN;
        this.setState(state);
    }

    render() {
        return (
            <Container className="control-container">
                <Container className="control-button-container">
                    <Row>
                        <Col></Col>
                        <Col md="auto">
                            <Button variant="success" size="lg" onClick={this.props.onClickAdvance}>
                                <FontAwesomeIcon icon={faPlayCircle}/>
                            </Button>
                        </Col>
                        <Col></Col>
                    </Row>
                    <Row>
                        <Col md="auto">
                            <Button variant="success" size="lg" onClick={this.props.onClickRotateLeft}>
                                <FontAwesomeIcon icon={faUndoAlt}/>
                            </Button>
                        </Col>
                        <Col></Col>
                        <Col md="auto">
                            <Button variant="success" size="lg" onClick={this.props.onClickRotateRight}>
                                <FontAwesomeIcon icon={faRedoAlt}/>
                            </Button>
                        </Col>
                    </Row>
                    <Row>
                        <Container className="step-settings-container">
                            <form>
                                <label htmlFor="step">Step Setting:</label>
                                <input type="number" id="step" name="step" 
                                    min="1"
                                    value={this.state.stepSetting}
                                    onChange={(event)=>this.handleInputChange(event)}
                                    onBlur={this.props.onStepSettingChange}
                                />
                            </form>
                        </Container>
                    </Row>               
                </Container>
                <Container className="user-terminate-container">
                    <Row>
                        <Col md={{ span: 9, offset: 2 }}>
                            <Button variant="danger" onClick={this.props.onUserTerminate}>
                                End Simulation
                            </Button>
                        </Col>
                    </Row>
                </Container>
            </Container> 
        );
    }
    
}
  
export default ControlComponent;