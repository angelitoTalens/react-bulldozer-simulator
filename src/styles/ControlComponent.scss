.control-container{
    width: 270px;
    border: 1px solid #999;
    border-radius: 30px;
    padding-bottom: 20px;
    padding-top: 35px;

}

.control-button-container {
    width: 200px;
    border: 1px solid #999;
    border-radius: 10px;
    padding-bottom: 20px;
    padding-top: 20px;

}

.user-terminate-container {
    padding-top: 20px;
}