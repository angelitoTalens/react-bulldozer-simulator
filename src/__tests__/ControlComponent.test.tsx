import React from 'react';
import renderer from 'react-test-renderer';

import ControlComponent from '../ControlComponent';

test ("Test ControlComponent", ()  => {
    const component = renderer.create(
        <ControlComponent
            onClickRotateLeft={()=>{}}
            onClickRotateRight={()=>{}}
            onClickAdvance={()=>{}}
            onUserTerminate={()=>{}}
            onStepSettingChange={()=>{}}
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});