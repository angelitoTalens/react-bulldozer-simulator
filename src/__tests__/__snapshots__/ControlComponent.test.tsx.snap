// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ControlComponent 1`] = `
<div
  className="control-container container"
>
  <div
    className="control-button-container container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      />
      <div
        className="col-md-auto"
      >
        <button
          className="btn btn-success btn-lg"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play-circle fa-w-16 "
            data-icon="play-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm115.7 272l-176 101c-15.8 8.8-35.7-2.5-35.7-21V152c0-18.4 19.8-29.8 35.7-21l176 107c16.4 9.2 16.4 32.9 0 42z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
      <div
        className="col"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-auto"
      >
        <button
          className="btn btn-success btn-lg"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-undo-alt fa-w-16 "
            data-icon="undo-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M255.545 8c-66.269.119-126.438 26.233-170.86 68.685L48.971 40.971C33.851 25.851 8 36.559 8 57.941V192c0 13.255 10.745 24 24 24h134.059c21.382 0 32.09-25.851 16.971-40.971l-41.75-41.75c30.864-28.899 70.801-44.907 113.23-45.273 92.398-.798 170.283 73.977 169.484 169.442C423.236 348.009 349.816 424 256 424c-41.127 0-79.997-14.678-110.63-41.556-4.743-4.161-11.906-3.908-16.368.553L89.34 422.659c-4.872 4.872-4.631 12.815.482 17.433C133.798 479.813 192.074 504 256 504c136.966 0 247.999-111.033 248-247.998C504.001 119.193 392.354 7.755 255.545 8z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
      <div
        className="col"
      />
      <div
        className="col-md-auto"
      >
        <button
          className="btn btn-success btn-lg"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-redo-alt fa-w-16 "
            data-icon="redo-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256.455 8c66.269.119 126.437 26.233 170.859 68.685l35.715-35.715C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.75c-30.864-28.899-70.801-44.907-113.23-45.273-92.398-.798-170.283 73.977-169.484 169.442C88.764 348.009 162.184 424 256 424c41.127 0 79.997-14.678 110.629-41.556 4.743-4.161 11.906-3.908 16.368.553l39.662 39.662c4.872 4.872 4.631 12.815-.482 17.433C378.202 479.813 319.926 504 256 504 119.034 504 8.001 392.967 8 256.002 7.999 119.193 119.646 7.755 256.455 8z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="step-settings-container container"
      >
        <form>
          <label
            htmlFor="step"
          >
            Step Setting:
          </label>
          <input
            id="step"
            min="1"
            name="step"
            onBlur={[Function]}
            onChange={[Function]}
            type="number"
            value={1}
          />
        </form>
      </div>
    </div>
  </div>
  <div
    className="user-terminate-container container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-9 offset-md-2"
      >
        <button
          className="btn btn-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          End Simulation
        </button>
      </div>
    </div>
  </div>
</div>
`;
